### PR TITLE
Add nn functional tests in JIT

### PR DESCRIPTION
**Pull request by wanchaol**
The PR is the first step to integrate torch.nn library with JIT. It adds the tests for nn functional interfaces in trace/script mode, and tries to find out the different between torch.nn.functional ops and the ATen ops, to see the work need to be done in order to support a full set of nn functional in script mode. 

Some statistics in summary:

- Totally 84 useful functions in torch.nn.functional (the number does not include helper funcs and deprecated funcs in torch.nn.functional). 

- 7 functions/ops does not support higher gradient, so just excluded from the whole test.

- 36 functions is different with the Aten op for different reasons. Among those 36 functions, bunch of them (roughly around 10-15) are just naming difference and simple transformation using other ops inside the function.

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -6139,5 +6301,10 @@ def fn(*inputs, **kwargs):
 for test in method_tests:
     add_test(*test)
 
+for test in nn_functional_tests:
+    # print(*test)
```
Comment (SsnL) [inline test/test_jit.py]: nit: print

**Comment by ezyang** [comment]
Is this still WIP?

File: test/test_jit.py [hunk 2]
```diff
@@ -6201,30 +6381,21 @@ def check(name):
...
                 args_variable, kwargs_variable = create_input(args, requires_grad=not is_inplace, call_kwargs=kwargs)
+
                 self_tensor = deepcopy(self_variable.data)
                 args_tensor = deepcopy(unpack_variables(args_variable))
-                output_variable = getattr(self_variable, name)(*args_variable, **kwargs_variable)
 
-                def fn(*inputs, **kwargs):
-                    output = getattr(inputs[0], name)(*inputs[1:], **kwargs)
-                    return output_process_fn(output)
+                if test_type == 'nn_op':
```
Comment (zdevito) [inline test/test_jit.py]: I think you should just create a separate `add_nn_test` function. Refactor the common parts into helper functions The huge if statement here is a good sign that these do not do the same thing.

**Review by zdevito — APPROVED** [review]
Looks good now
